server/eventlog.go

```diff
@@ -14,13 +15,19 @@ import (
 
 // ParseMachineState parses a raw event log and replays the parsed event
 // log against the given PCR values. It returns the corresponding MachineState
-// containing the events verified by particular PCR indexes/digests. An error is
-// returned if the replay for any PCR index does not match the provided value.
+// containing the events verified by particular PCR indexes/digests. It returns
+// an error if the replay for any PCR index does not match the provided value.
+//
+// The returned MachineState may be a partial MachineState where fields can be
+// the zero value. In this case, an error of type MachineStateError will be
+// returned. Callers can inspect individual parsing errors by examining
+// `MachineStateError.Errors`.
 //
 // It is the caller's responsibility to ensure that the passed PCR values can be
 // trusted. Users can establish trust in PCR values by either calling
 // client.ReadPCRs() themselves or by verifying the values via a PCR quote.
 func ParseMachineState(rawEventLog []byte, pcrs *tpmpb.PCRs) (*pb.MachineState, error) {
+	var errors []error
 	events, err := parseReplayHelper(rawEventLog, pcrs)
 	if err != nil {
 		return nil, err
```

Comment:
Can this error be added to the groupedErrors as well? So the function can just return a groupedError

Reply:
`errors` is used to create the returned `GroupedErrors`.

Reply:
Oh I mean the `err` on line 31 returned by `parseReplayHelper`, can we add that to the `GroupedErrors` `errors`?

Reply:
Ohhhh I see. This is a great catch, thanks! yes, we want this wrapped in a GroupedError b/c that's what this func returns. but likely by itself if there is a complete failure to parse.